Client/Frontend/Browser/HomePanel/NTPBackgroundDataSource.swift

```diff
@@ -204,6 +204,7 @@ extension NTPBackgroundDataSource: PreferencesObserver {
         if sponsoredPref.key == key {
             if sponsoredPref.value {
                 // Enabled? -> issue download
+                Preferences.NTP.ntpCheckDate.value = nil
```

Comment:
without it, after reenabling the setting user needs to wait for another NTP fetch tick(~5 hours)